sekoiaio_connector.py

```diff
@@ -0,0 +1,439 @@
+# File: sekoiaio_connector.py
+#
+# Copyright (c) 2023 SEKOIA.IO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+import json
+
+# Phantom App imports
+import phantom.app as phantom
+# Useful libraries
+import requests
+from bs4 import BeautifulSoup
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+# Usage of the consts file is recommended
+import sekoiaio_consts as consts
+
+
+class RetVal(tuple):
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class SekoiaioConnector(BaseConnector):
+    def __init__(self):
+        super(SekoiaioConnector, self).__init__()
+        self._state = None
+        self._base_url = None
+        self.api_key = None
+
+    def initialize(self):
+        """
+        This's an optinal method but we will use it to load
+        both state and configuration
+        """
+        self._state = self.load_state()
+        config = self.get_config()
+        self._base_url = config["base_url"]
+        self.api_key = config["api_key"]
```

Comment:
Any reason why this one is public and the other two fields are protected?

Reply:
Good question, It was suggested by Splunk SOAR when generating files. So I thought that it was for best practices in splunk. I change them all to public.